python/hsml/engine/model_engine.py

```diff
@@ -88,40 +88,73 @@ def _upload_additional_resources(self, model_instance):
             model_instance.model_schema = None
         return model_instance
 
-    def _copy_hopsfs_model(self, existing_model_path, model_version_path):
+    def _copy_or_move_hopsfs_model(
+        self,
+        existing_model_path,
+        model_version_path,
+        keep_original_files,
+        update_upload_progress,
+    ):
         # Strip hdfs prefix
         if existing_model_path.startswith("hdfs:/"):
             projects_index = existing_model_path.find("/Projects", 0)
             existing_model_path = existing_model_path[projects_index:]
 
+        n_dirs, n_files = 0, 0
         for entry in self._dataset_api.list(existing_model_path, sort_by="NAME:desc")[
             "items"
         ]:
             path = entry["attributes"]["path"]
             _, file_name = os.path.split(path)
-            self._dataset_api.copy(path, model_version_path + "/" + file_name)
+            if keep_original_files:
+                self._dataset_api.copy(path, model_version_path + "/" + file_name)
+            else:
+                self._dataset_api.move(path, model_version_path + "/" + file_name)
+            if "." in path:
+                n_files += 1
+            else:
+                n_dirs += 1
+            update_upload_progress(n_dirs=n_dirs, n_files=n_files)
 
     def _upload_local_model(
-        self, local_model_path, model_version, dataset_model_name_path
+        self,
+        local_model_path,
+        dataset_model_name_path,
+        update_upload_progress,
     ):
-        archive_out_dir = None
-        uploaded_archive_path = None
-        try:
-            archive_out_dir = tempfile.TemporaryDirectory(dir=os.getcwd())
-            archive_path = util.compress(
-                archive_out_dir.name, str(model_version), local_model_path
+        n_dirs, n_files = 0, 0
+        for root, dirs, files in os.walk(local_model_path):
```

Comment:
may be a short comment of the logic of upload would help

Reply:
Done